content/300-guides/500-other/900-advanced-database-tasks/06-typesafe-raw-sql/01-safeql-gis.mdx

```diff
@@ -0,0 +1,128 @@
+
+Our example will be using [PostGIS](https://postgis.net/) and PostgreSQL, but is applicable to other features and database engines.
+
+## Prerequisites
+
+To follow this guide, we recommend:
+
+- an existing [PostgreSQL](https://www.postgresql.org/) database
+- an existing Prisma project
+
+## What is SafeQL?
+
+[SafeQL](https://safeql.dev/) is an ESLint plugin that allows for advanced linting and type safety within raw SQL queries. After setup, SafeQL works with Prisma `$queryRaw` and `$executeRaw` to provide type safety when raw queries are required.
+
+## Geographic data support in Prisma
+
+At this time geographic data, and specifically PostGIS, are not supported by Prisma. Models that have geographic data columns will not have the benefit of type safety, nor will be able to use mutating operations like `create` and `update`.
+
+Prisma does support operations using `$queryRaw` and `$executeRaw` and we will be using these in order to create a Prisma Client extension with SafeQL to improve the usage of geographic data in Prisma.
+
+## 1. Set up Prisma for use with PostGIS
+
+If you haven't already, enable the `postgresqlExtensions` Preview feature and add the `postgis` PostgreSQL extension in your Prisma schema:
+
+```prisma highlight=3,8;add
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["postgresqlExtensions"]
+}
+
+datasource db {
+  provider   = "postgresql"
+  url        = env("DATABASE_URL")
+  extensions = [postgis]
+}
+```
+
+Then, create a migration using `npx prisma migrate dev` in order to enable the extension. The migration file should look like the following:
+
+```sql
+
+-- CreateExtension
+CREATE EXTENSION IF NOT EXISTS "postgis";
+
+```
+
+Make sure that the migration is applied!
+
+## 2. Create a new model that uses a geographic data column
+
+After the migration is applied, you can now add a new model that has a GIS column. For our example we'll be using the following `PointOfInterest` model.
+
+```prisma
+model PointOfInterest {
+  id       Int                                   @id @default(autoincrement())
+  name     String
+  location Unsupported("geography(Point, 4326)")
+}
+```
+
+You'll notice that `location` is marked as `Unsupported`. This means that we lose a lot of the benefits of Prisma when working with `PointOfInterest`. We'll be using [SafeQL](https://safeql.dev/) to fix this.
+
+Create a new migration with `npx prisma migrate dev` and apply it to add the `PointOfInterest` table to our database.
+
+## 3. Integrate SafeQL
+
+SafeQL is easily integrated with Prisma. Follow [SafeQL's integration guide](https://safeql.dev/compatibility/prisma.html) in order to lint `$queryRaw` and `$executeRaw` Prisma calls.
+
+<Admonition>
+
+Note that for SafeQL to run it will need to have access to your database. We recommend passing the same environment variable or connection string that is used by Prisma.
+
+</Admonition>
```

Comment:
I think this is a little hand-wavy. I got stuck here unsure exactly how to set up/ integrate SafeQL 🙂

Let's spell out the instructions here.

Reply:
Additionally, I think it would be useful to helpful to have an example repository with a minimal setup we could point users to that they can refer to if they get stumped 😄